sympy/utilities/lambdify.py

```diff
@@ -742,8 +742,35 @@ def _lambdifygenerated(x):
     argument is not provided, ``lambdify`` creates functions using the NumPy
     and SciPy namespaces.
     """
-    from sympy.core.symbol import Symbol
+    from sympy.core.symbol import Symbol, Dummy
     from sympy.core.expr import Expr
+    from sympy.matrices import DeferredVector
+    from sympy.core.basic import Basic
+    from sympy.core.function import Derivative, Function
+    if dummify is not False:
+        if not iterable(args, exclude=DeferredVector):
+            it = False
+            args = [args]
+        else:
+            it = True
+        reps = {}
+        for a in args:
+            if isinstance(a, (Function, Derivative)):
+                d = Dummy()
+                reps.update({a: d})
+        if reps:
+            args = [reps.get(a, a) for a in args]
+            if not it:
+                args = args[0]
+            if iterable(expr):
+                if isinstance(expr, DeferredVector):
+                    expr = expr.xreplace(reps)
+                else:
+                    expr = [i.xreplace(reps) for i in expr]
+            elif isinstance(expr, Basic):
+                expr = expr.xreplace(reps)
+            return lambdify(args, expr, modules=modules, printer=printer,
+                use_imps=use_imps, dummify=dummify, cse=cse)
```

Comment:
Where did all of this come from? I thought that the issue had been fixed without these changes. Why are they needed?

Reply:
When cse=True, the test result I showed above failed. The docstring says that these things should be handled by default but the `dummify=False` flag doesn't allow for that. Now it does.

Reply:
These are handled in `lambdastr` but were not handled in `lambdify`. If you have a better suggestion, please make the change.

Reply:
If this is not immediately connected to fixing the regression introduced in the 1.11 release then can it go in a separate PR? I was hoping to backport this one for a bugfix release.

Reply:
In the discussion here or at the issue, it was suggested to set `cse=True` to see what other bugs were lurking still. Another such bug was discovered. I don't know what is involved with backporting and why you would want to backport one bugfix but not another. If you want to take only @moorepants pants initial fix, can't you just cherry pick the first three commits?

Reply:
> I don't know what is involved with backporting and why you would want to backport one bugfix but not another

The point is just to keep the change as targeted as possible. Bugs are fixed (and reintroduced) all the time in SymPy so it would be impractical to backport all fixes. More importantly every backported change risks introducing new bugs so we want to keep those changes minimal.

Of course it does make sense to fix any other bugs on master but the test failure you highlighted above is not a regression: that problem has existed since the cse flag was added. The fix that you have just added looks surprisingly complex (i.e. high risk for new bugs). It also makes this PR more complicated which delays getting it merged.

The fix also looks fishy to me. Why do we need to special case things like `DeferredVector`? Would it not just make sense just to turn all the args into dummies? (You don't need to answer those questions but please understand that adding these extra changes makes it more difficult to get this PR merged and so delays getting a fix released.)

> can't you just cherry pick the first three commits?

It can be confusing to keep track of what changes are on what branch when there are multiple branches. I would prefer that the PR was merged to master and then backported in full. Actually I was just about to squash the commits so I could backport it as a single commit.

Thank you for fixing the test so that it doesn't depend on NumPy but this PR was opened with the simple purpose being to fix the regression (hence the title) so can changes not related to that go somewhere else?

Reply:
I opened #23975 to separate the basic fix.